hack/v1migration/delete.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+WORKDIR=${1:?must specify working directory}
+echo "Using workdir: $WORKDIR"
+
+case "${2}" in
+"--dry-run" )
+  dry_run="--dry-run"
+  ;;
+"" )
+  # When doing a dry run, allow the script to continue after errors. Operations such as deleting a CRD are likely
+  # to fail since the CRs are not being deleted in a dry run.
+  set -e
+  ;;
+* )
+  echo "Only valid flag is \"--dry-run\""
+  exit 1
+esac
+
+if [[ -z $dry_run ]]
+then
+  source "$(dirname $0)/verify_scaledown.sh"
+  verify_all_scaled_down
+fi
+
+mkdir -p "$WORKDIR"
+
+if [[ -z $dry_run && \
+      "$(ls "$WORKDIR" 2>/dev/null)" ]]
+then
+  echo "WARNING: workdir ($WORKDIR) is not empty."
+  echo "Files in workdir may be overwritten."
+  read -r -p "Do you wish to proceed? [y/N] " response
+  if [[ ! "$response" =~ ^([yY]([eE][sS])?)$ ]]
+  then
+    exit 1
+  fi
+fi
+
+readarray -t HIVE_TYPES <<< \
+  "$(oc get crd -o json | \
+    jq -r '.items[].spec | select((.group="hive.openshift.io") and (.version="v1alpha1")) | .names.plural')"
+
+for t in "${HIVE_TYPES[@]}"
+do
+	echo "Storing all ${t} in ${WORKDIR}/${t}.json"
+	oc get "${t}.hive.openshift.io" -A -o json | jq .items[] > "${WORKDIR}/${t}.json"
+done
```

Comment:
Hit some unexpected things here running ./hack/v1migration/delete.sh migration --dry-run:

```
Using workdir: migration                                                                                                                                                                                            
Storing all alertmanagers in migration/alertmanagers.json                                                                                                                                                           
error: the server doesn't have a resource type "alertmanagers"                                                                                                                                                      
Storing all apiservers in migration/apiservers.json                                                                                                                                                                 
error: the server doesn't have a resource type "apiservers"                                                                                                                                                         
Storing all authentications in migration/authentications.json                                                                                                                                                       
error: the server doesn't have a resource type "authentications"                                                                                                                                                    
Storing all authentications in migration/authentications.json                                                                                                                                                       
error: the server doesn't have a resource type "authentications"                                                                                                                                                    
Storing all baremetalhosts in migration/baremetalhosts.json                                                                                                                                                         
error: the server doesn't have a resource type "baremetalhosts"                                                                                                                                                     
Storing all builds in migration/builds.json                                                                                                                                                                         
error: the server doesn't have a resource type "builds"                                                                                                                                                             
Storing all catalogsourceconfigs in migration/catalogsourceconfigs.json                                                                                                                                             
error: the server doesn't have a resource type "catalogsourceconfigs"                                                                                                                                               
Storing all catalogsources in migration/catalogsources.json                                                                                                                                                         
error: the server doesn't have a resource type "catalogsources"                                                                                                                                                     
Storing all checkpoints in migration/checkpoints.json                                                                                                                                                               
Storing all clusterautoscalers in migration/clusterautoscalers.json                                                                                                                                                 
error: the server doesn't have a resource type "clusterautoscalers"                                                                                                                                                 
Storing all clusterdeployments in migration/clusterdeployments.json                                                                                                                                                 
Storing all clusterdeprovisionrequests in migration/clusterdeprovisionrequests.json        
```

Similarly later on:

```
Deleting alertmanagers objects                                                                                                                                                                                      
Deleting alertmanagers.hive.openshift.io CRD                                                                                                                                                                        
Deleting apiservers objects                                                                                                                                                                                         
Deleting apiservers.hive.openshift.io CRD                                                                                                                                                                           
Deleting authentications objects                                                                                                                                                                                    
Deleting authentications.hive.openshift.io CRD                                                                                                                                                                      
Deleting authentications objects                                                                                                                                                                                    
Deleting authentications.hive.openshift.io CRD                                                                                                                                                                      
Deleting baremetalhosts objects                                                                                                                                                                                     
Deleting baremetalhosts.hive.openshift.io CRD                                                                                                                                                                       
Deleting builds objects                                                                                                                                                                                             
Deleting builds.hive.openshift.io CRD                                                                                                                                                                               
Deleting catalogsourceconfigs objects                                                                                                                                                                               
Deleting catalogsourceconfigs.hive.openshift.io CRD                                                                                                                                                                 
Deleting catalogsources objects                                                                                                                                                                                     
Deleting catalogsources.hive.openshift.io CRD                                                                                                                                                                       
Deleting checkpoints objects                                                                                                                                                                                        
Deleting checkpoints.hive.openshift.io CRD                                                                                                                                                                          
Deleting clusterautoscalers objects                                                                                                                                                                                 
Deleting clusterautoscalers.hive.openshift.io CRD                                                                                                                                                                   
Deleting clusterdeployments objects                                                                                                                                                                                 
INFO[0000] deleted object                                apiVersion=hive.openshift.io/v1alpha1 kind=ClusterDeployment name=dgoodwin1 namespace=dgoodwin1                                                            
INFO[0000] deleted object                                apiVersion=hive.openshift.io/v1alpha1 kind=ClusterDeployment name=dgoodwin2 namespace=dgoodwin2           
```

Lot of output snipped there but it looks like it tried to operator on all CRD types rather than just hive.

Reply:
Per slack we'll go back to hardcoded list of hive CRDs.